criteria.c

```diff
@@ -27,18 +27,34 @@ struct mako_criteria *create_criteria(struct mako_config *config) {
 	return criteria;
 }
 
+void free_cond(struct mako_condition *cond) {
+	switch(cond->operator) {
+		case OP_EQUALS:
+		case OP_NOT_EQUALS:
+			free(cond->value);
+			return;
+		case OP_REGEX_MATCHES:
+			regfree(&cond->pattern);
+			return;
+		case OP_NONE:
+		case OP_TRUTHY:
+		case OP_FALSEY:
+		default:
```

Comment:
Could we drop this `default`? That way, the compiler will warn when we're missing a case.